### PR TITLE
Adding support for Premium V3 tier

### PR DESCRIPTION
**Pull request by vinisoto**
## Description

Adding support for new pricing tier Premium V3

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs [hunk 1]
```diff
@@ -35,8 +35,8 @@ public class NewAzureAppServicePlanCmdlet : AppServicePlanBaseCmdlet
         [ValidateNotNullOrEmpty]
         public string Location { get; set; }
 
-        [Parameter(Position = 3, Mandatory = false, HelpMessage = "The App Service plan tier. Allowed values are [Free|Shared|Basic|Standard|Premium|PremiumV2]")]
-        [PSArgumentCompleter("Free", "Shared", "Basic", "Standard", "Premium", "PremiumV2", "Isolated", "PremiumContainer")]
+        [Parameter(Position = 3, Mandatory = false, HelpMessage = "The App Service plan tier. Allowed values are [Free|Shared|Basic|Standard|Premium|PremiumV2|PremiumV3]")]
+        [PSArgumentCompleter("Free", "Shared", "Basic", "Standard", "Premium", "PremiumV2", "PremiumV3", "Isolated", "PremiumContainer")]
```
Comment (jvano) [inline src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs:39]: >, "PremiumContainer" [](start = 115, length = 20)

we won't GA PremiumContainer SKU and we should remove it

**Comment by jvano** [comment]
>             }

same here

---
Refers to: src/Websites/Websites/Utilities/CmdletHelpers.cs:299 in d43001f. [](commit_id = d43001f05c8697467bddaab56621f37be6836f03, deletion_comment = False)

**Comment by jvano** [comment]
>             }

same here

---
Refers to: src/Websites/Websites/Utilities/CmdletHelpers.cs:330 in d43001f. [](commit_id = d43001f05c8697467bddaab56621f37be6836f03, deletion_comment = False)

File: src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs [hunk 2]
```diff
@@ -73,9 +73,10 @@ public class NewAzureAppServicePlanCmdlet : AppServicePlanBaseCmdlet
 
         public override void ExecuteCmdlet()
         {
-            if (HyperV.IsPresent && Tier != "PremiumContainer")
+            if (HyperV.IsPresent && 
+                (Tier != "PremiumContainer" && Tier != "PremiumV3"))
```
Comment (jvano) [inline src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs:77]: >(Tier != "PremiumContainer" &&  [](start = 16, length = 31)

same here

**Review by jvano — APPROVED** [review]
:shipit:

File: src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs (hunk 1 above)
Comment (vinisoto) [reply src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs:39]: Yes. Will remove it in next PR.

File: src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs (hunk 2 above)
Comment (vinisoto) [reply src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs:77]: Yes. Will remove it in next PR.

Comment (pagariyaalok) [inline src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs:77]: >Tier != "PremiumV3" [](start = 47, length = 19)

should we worry about string case here ?

Comment (vinisoto) [reply src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs:77]: We validate specifically against the values PSArgumentCompleter for Tier (that is the general convention in Azure PowerShell).

**Review by pagariyaalok — APPROVED** [review]
:shipit:

**Review by msJinLei — CHANGES_REQUESTED** [review]
@vinisoto Please also update help files for the changed cmdlets

**Comment by vinisoto** [comment]
@msJinLei Help files for New-AzWebApp and Set-AzWebApp don't require changes since the values for the Tier parameter are currently not specified there. For example:

### -Tier
Tier

```yaml
Type: System.String
Parameter Sets: (All)
Aliases:

Required: False
Position: 3
Default value: Free
Accept pipeline input: False
Accept wildcard characters: False
```

See the files for reference: 
https://github.com/Azure/azure-powershell/blob/master/src/Websites/Websites/help/New-AzAppServicePlan.md
https://github.com/Azure/azure-powershell/blob/master/src/Websites/Websites/help/Set-AzAppServicePlan.md

**Review by msJinLei — APPROVED** [review]

**Pull request merged**